app/components/text_analysis/TextAnalysis.tsx

```diff
@@ -0,0 +1,90 @@
+import { useState } from "react"
+import getSentimentAnalysis from "../../services/text_analysis/get-sentiment-analysis"
+import styles from "../../styles/components/text-analysis.module.scss"
+import { TextAnalysisProps } from "../../constants/types/components_props/types"
+
+const TextAnalysis = ({
+  textCta,
+  inputCta,
+  textAreaPlaceholder,
+  inputPlaceholder,
+  buttonText,
+}: TextAnalysisProps) => {
+  const [userMessageContext, setUserMessageContext] = useState("")
+  const [userMessage, setUserMessage] = useState("")
+  const [textAnalysisResponse, setTextAnalysisResponse] = useState("")
+  const [errorMessage, setErrorMessage] = useState("")
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault()
+    if (!userMessage.trim() || !userMessageContext.trim()) {
+      setErrorMessage("Please fill out both fields.")
+      return
+    }
+
+    try {
+      const response = await getSentimentAnalysis(userMessage, userMessageContext)
+      const data = await response.json()
+      setTextAnalysisResponse(data.content)
+      console.log(data)
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  function handleValidation(event: React.FormEvent<HTMLInputElement>) {
+    const input = event.target as HTMLInputElement
+    input.setCustomValidity("This field is required.")
+  }
+
+  function handleInput(event: React.FormEvent<HTMLInputElement>) {
+    const input = event.target as HTMLInputElement
+    input.setCustomValidity("")
+  }
+
+  return (
+    <div className={styles.container}>
+      <form onSubmit={handleSubmit}>
+        <label htmlFor="user-message">{textCta}</label>
+        <input
+          className={styles.user_message}
+          id="user-message"
+          type="text"
+          placeholder={textAreaPlaceholder}
+          value={userMessage}
+          onChange={(event) => setUserMessage(event.target.value)}
+          aria-label={textAreaPlaceholder}
+          aria-invalid={!userMessage.trim()}
+          aria-describedby="user-message-error"
+          onInvalid={handleValidation}
+          onInput={handleInput}
+          required
```

Comment:
In line 20 there is a validation when the user clicked button submit and field input is empty, Could you valid why there are two valids, the first in line 20 and other this line?

Reply:
you are right, these validations might be redundant

validation from line 20 was removed, individual validation message was kept on inputs